python_modules/libraries/dagster-tableau/dagster_tableau/resources.py

```diff
@@ -452,6 +489,17 @@ def fetch_tableau_workspace_data(
                                         properties=published_data_source_data,
                                     )
                                 )
+                        if is_publish_data_source_absent:
```

Comment:
Here as well, you could test the list itself instead of using an additional flag:

```suggestion
                        if not published_data_source_list:
```